src/System.Management.Automation/engine/remoting/client/remoterunspace.cs

```diff
@@ -2610,6 +2794,42 @@ private void SetIsActive(int breakpointCount)
             }
         }
 
+        private T InvokeRemoteBreakpointFunction<T>(string functionName, Dictionary<string, object> parameters)
+        {
+            CheckForValidateState();
+
+            using (PowerShell ps = GetNestedPowerShell())
+            {
+                ps.AddCommand(functionName);
+                foreach (var parameterName in parameters.Keys)
+                {
+                    ps.AddParameter(parameterName, parameters[parameterName]);
+                }
+
+                Collection<PSObject> output = ps.Invoke<PSObject>();
+
+                // If an error exception is returned then throw it here.
+                if (ps.ErrorBuffer.Count > 0)
+                {
+                    Exception e = ps.ErrorBuffer[0].Exception;
+                    if (e != null)
+                    {
+                        throw e;
+                    }
+                }
+
+                foreach (var item in output)
+                {
+                    if (item?.BaseObject is T)
```

Comment:
I don't think I can use that pattern here, because `T` may be a class, or it may be a value type (e.g. `bool`), which means I cannot place a type constraint on it, which in turn means I can't use `as` with it.

Reply:
Hmm, Ok.  Let's leave it as is then.